subprojects/core/src/main/java/org/gradle/internal/classpath/CopyingClasspathFileTransformer.java

```diff
@@ -39,8 +39,8 @@ public File transform(File source, CompleteFileSystemLocationSnapshot sourceSnap
             // Directories are ok to use outside the cache
             return source;
         }
-        for (AdditiveCache cache : additiveCaches) {
-            for (File root : cache.getAdditiveCacheRoots()) {
+        for (GlobalCache cache : globalCaches) {
+            for (File root : cache.getGlobalCacheRoots()) {
                 if (source.toPath().startsWith(root.toPath())) {
```

Comment:
Why don't we inject `GlobalCacheLocations` here instead? All we need to know here is whether or not a particular path is in a global cache.

Reply:
Doing that now.